src/ucp/api/ucp.h

```diff
@@ -1172,6 +1213,55 @@ typedef struct ucp_worker_params {
 } ucp_worker_params_t;
 
 
+/**
+ * @ingroup UCP_ENDPOINT
+ * @brief UCP performance endpoint attributes.
+ *
+ * The structure defines the attributes which characterize
+ * the performance of a particular endpoint.
+ */
+typedef struct {
+    /**
+     * Mask of valid fields in this structure, using bits from
+     * @ref ucp_ep_perf_attr_field.
+     * Fields not specified in this mask will be ignored.
+     * Provides ABI compatibility with respect to adding new fields.
+     */
+    uint64_t          field_mask;
+
+    /**
+     * Message size to use for determining performance.
+     * This field must be initialized by the caller.
+     */
+    size_t            message_size;
+
+    /**
+     * Local memory type to use for determining performance.
+     * This field must be initialized by the caller.
+     */
+    ucs_memory_type_t local_memory_type;
+
+    /**
+     * Remote memory type to use for determining performance.
+     * Relevant only for operations that have remote memory access.
+     * This field must be initialized by the caller.
+     */
+    ucs_memory_type_t remote_memory_type;
+
+    /**
+     * Operation to report performance for.
+     * This field must be initialized by the caller.
+     */
+    uct_ep_op_t       operation;
+
+    /**
+     * Bandwidth this endpoint is able to provide.
+     * This field is set by the UCP layer.
+     */
+    double            bandwidth;
```

Comment:
instead of bandwidth: we can provide estimated to deliver the message
this will encompass both bw and latency
user can calc size/time to get bw

Reply:
Do you mean a time estimate (i.e `bw / size`) ?